File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteMiddle(ListNode* head) {
        if(head==NULL || head->next==NULL)
            return NULL;
        
        ListNode* node1 = head, *node2 = head;
        int size=0;
        while(node1!=NULL){
            size++;
            node1=node1->next;
        }
        
        int middle = size/2;
        
        for(int i=0;i<middle-1;i++){
            node2=node2->next;
        }
        node2->next=node2->next->next;
        return head;
    }
};

File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteMiddle(ListNode head) {
        
        if(head == null || head.next == null){
            return null;
        }
        
        ListNode temp = head;
        
        int cnt = 0;
        while(temp!=null){
            temp = temp.next;
            cnt++;
        }
        
        int mid = cnt/2;
        System.out.println(mid);
        
        
        temp = head;
        
        while(temp!=null){
            
            mid--;
            
            if(mid == 0){
                 temp.next = temp.next.next;
                break;
            }
            temp = temp.next;
        }
        
        return head;
    }
}